spec/presenters/work_provenance_display_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
```

Comment:
There are only tests of the `split_provenance` method, which isn't actually the public API. 

We need at least minimal test(s) of the `display` method that is actually the public API.  1. Test that for a typical "happy path" case, it produces the HTML we expect. 2. Test that for `nil` input, it returns an empty string. 

See existing view model tests for how to use Nokogiri to parse the output, and then use capybara testers `have_selector` or `have_content` to test output. Or you can just use nokogiri manually. We can discuss more if unclear. 

An alternate approach would be trying to use the actual `#display` output for _all_ tests. Less convenient cause you have to deal with the HTML when you really just want to test "did it split the summary from the notes right?", but avoiding testing on what's really an internal method. 

Or perhaps a medium ground, your existing tests could be written in terms of the `provenance_summary` and `provenance_notes`  methods to test exactly what you want without having to deal with HTML apparatus, but not use the more internal `split` method -- which could then be marked private. Right now, `provenance_summary` and `provenance_notes` are only _available_ after you call `display`, so writing the tests this way might require some refactoring there, so calling those methods would trigger splitting and population of the ivars. 

```ruby
# lazily make sure split happened and ivars are filled
def populate_parts
   unless @populate_parts
      @provenance_summary, @provenance_notes = split_provenance
      @populate_parths = true
   end
end

  def provenance_summary
    populate_parts
    DescriptionDisplayFormatter.new(@provenance_summary).format
  end
```

That would also have the benefit of allowing this class to be easily used by a caller that wants to split provenance notes and summary for some non-HTML purpose, and won't be calling display. 

But these are just an ideas, arguably existing tests can stand as they are -- but still needs at least minimal tests of `#display`

The non-nil `#display` test should definitely use *real sample data* that looks like real data, not placeholder text like `"provenance metadata"`.

Reply:
Good call. I used nokogiri to parse the template output. Added a couple span tags to make testing easier.